src/aap_eda/api/serializers/activation.py

```diff
@@ -129,6 +131,12 @@ def create(self, validated_data, user):
         validated_data["rulebook_rulesets"] = rulebook.rulesets
         validated_data["git_hash"] = rulebook.project.git_hash
         validated_data["project_id"] = rulebook.project.id
+        if validated_data["decision_environment_id"]:
+            validated_data[
+                "decision_environment_name"
+            ] = models.DecisionEnvironment.objects.get(
```

Comment:
We already have that check [here](https://github.com/ansible/eda-server/blob/aea444d606828512e3994455447b010906a09830/src/aap_eda/api/views/activation.py#L37), which gets executed when we try to use `decision_environment_id` in validated_data. I thought it would be redundant to throw an exception here as well. Thoughts?

Reply:
Is that function `handle_activation_create_conflict` called because this: https://github.com/ansible/eda-server/pull/402/files#diff-b75687856dd6a2e0a5e5a1d5d322ed63ca815ecd7fbfbb6aadb25009b4d0e89bL126? so I think we need a similar check on decision environment.

Reply:
It gets called because of the IntegrityError on this [line](https://github.com/ansible/eda-server/pull/402/files#diff-b75687856dd6a2e0a5e5a1d5d322ed63ca815ecd7fbfbb6aadb25009b4d0e89bR140) when any of the 3 related object IDs are incorrect.